test/unit/product/product-component.js

```diff
@@ -605,44 +566,180 @@ describe('Product Component class', () => {
+          product.updateVariant('Size', 'large');
+          assert.equal(product.selectedVariant, selectedVariant);
+          variantForOptionsStub.restore();
+        });
+
+        describe('if variant exists', () => {
+          let variantForOptionsStub;
+
+          beforeEach(() => {
+            product = Object.defineProperty(product, 'variantExists', {
+              value: true,
+            });
+          });
+
+          afterEach(() => {
+            if (variantForOptionsStub.restore) {
+              variantForOptionsStub.restore();
+            }
+          });
+
+          it('caches image', () => {
+            variantForOptionsStub = sinon.stub(props.client.product.helpers, 'variantForOptions').returns({image: 'test-image'});
+            product.updateVariant('Size', 'large');
+            assert.equal(product.cachedImage, 'test-image');
+          });
+
+          it('sets the selected image to null if selected variant has an image', () => {
+            variantForOptionsStub = sinon.stub(props.client.product.helpers, 'variantForOptions').returns({image: 'test-image'});
+            product.updateVariant('Size', 'large');
+            assert.isNull(product.selectedImage);
+          });
+
+          it('sets the selected image to the first image in model if selected variant does not have an image', () => {
+            variantForOptionsStub = sinon.stub(props.client.product.helpers, 'variantForOptions').returns({});
+            product.updateVariant('Size', 'large');
+            assert.equal(product.selectedImage, product.model.images[0]);
           });
         });
+
+        it('sets selected image to the cached image in model if variant does not exist', () => {
+          product = Object.defineProperty(product, 'variantExists', {
+            value: false,
+          });
+          const expectedImage = {
+            id: 1,
+            src: `${rootImageURI}image-one.jpg`,
+          };
+          product.updateVariant('Size', 'large');
+          assert.deepEqual(product.selectedImage, expectedImage);
+        });
+
+        it('renders the view', () => {
+          product.updateVariant('Size', 'large');
+          assert.calledOnce(renderStub);
+        });
+
+        it('calls userEvent with updateVariant', () => {
+          product.updateVariant('Size', 'large');
+          assert.calledOnce(userEventStub);
+          assert.calledWith(userEventStub, 'updateVariant');
+        });
+
+        it('returns the updated option', () => {
+          const updatedOption = product.updateVariant('Size', 'large');
+          const expectedObj = {
+            name: 'Size',
+            selected: 'small',
+            values: [{value: 'small'}, {value: 'large'}],
+          };
+          assert.deepEqual(updatedOption, expectedObj);
+        });
       });
 
       describe('setDefaultVariant()', () => {
```

Comment:
I think we're missing tests around `selectedVariant`. Also, we should probably rewrite the descriptions to match the style of the new tests.